File: frontend/src/api/checkin.ts
import { apiClient } from './client'

export interface CheckinSettings {
  enabled: boolean
  content: string
  daily_reward: number
  extra_reward_4: number
  extra_reward_16: number
}

export interface CheckinRecord {
  id: number
  user_id: number
  checkin_date: string
  daily_reward: number
  extra_reward: number
  month_count: number
  consecutive_count: number
  extra_milestones: number[]
  checked_in_at: string
  created_at: string
  updated_at: string
}

export interface CheckinMonthSummary {
  year: number
  month: number
  today: string
  today_checked: boolean
  month_count: number
  consecutive_count: number
  consecutive_cycle_day: number
  days_in_month: number
  records: CheckinRecord[]
  next_extra_milestone?: number
}

export interface CheckinOverview {
  settings: CheckinSettings
  summary: CheckinMonthSummary
}

export interface CheckinResult {
  record: CheckinRecord
  summary: CheckinMonthSummary
  reward: number
  new_balance: number
}

export async function getOverview(): Promise<CheckinOverview> {
  const { data } = await apiClient.get<CheckinOverview>('/checkin')
  return data
}

export async function checkin(): Promise<CheckinResult> {
  const { data } = await apiClient.post<CheckinResult>('/checkin')
  return data
}

export const checkinAPI = {
  getOverview,
  checkin,
}

export default checkinAPI

File: backend/internal/service/checkin.go
package service

import "time"

// CheckinSettings 是用户侧签到运行时配置。
type CheckinSettings struct {
	Enabled       bool    `json:"enabled"`
	Content       string  `json:"content"`
	DailyReward   float64 `json:"daily_reward"`
	ExtraReward4  float64 `json:"extra_reward_4"`
	ExtraReward16 float64 `json:"extra_reward_16"`
}

// CheckinRecord 表示一次实际签到记录。
type CheckinRecord struct {
	ID               int64     `json:"id"`
	UserID           int64     `json:"user_id"`
	CheckinDate      time.Time `json:"checkin_date"`
	DailyReward      float64   `json:"daily_reward"`
	ExtraReward      float64   `json:"extra_reward"`
	MonthCount       int       `json:"month_count"`
	ConsecutiveCount int       `json:"consecutive_count"`
	ExtraMilestones  []int     `json:"extra_milestones"`
	CheckedInAt      time.Time `json:"checked_in_at"`
	CreatedAt        time.Time `json:"created_at"`
	UpdatedAt        time.Time `json:"updated_at"`
}

type CheckinMonthSummary struct {
	Year                int             `json:"year"`
	Month               int             `json:"month"`
	Today               string          `json:"today"`
	TodayChecked        bool            `json:"today_checked"`
	MonthCount          int             `json:"month_count"`
	ConsecutiveCount    int             `json:"consecutive_count"`
	ConsecutiveCycleDay int             `json:"consecutive_cycle_day"`
	DaysInMonth         int             `json:"days_in_month"`
	Records             []CheckinRecord `json:"records"`
	NextExtraMilestone  *int            `json:"next_extra_milestone,omitempty"`
}

type CheckinOverview struct {
	Settings CheckinSettings     `json:"settings"`
	Summary  CheckinMonthSummary `json:"summary"`
}

type CheckinResult struct {
	Record     CheckinRecord       `json:"record"`
	Summary    CheckinMonthSummary `json:"summary"`
	Reward     float64             `json:"reward"`
	NewBalance float64             `json:"new_balance"`
}
